types/react-navigation/index.d.ts

```diff
@@ -86,7 +87,7 @@ export type NavigationLeafRoute<Params> = {
    * Params passed to this route when navigating to it,
    * e.g. `{ car_id: 123 }` in a route that displays a car.
    */
-  params?: Params,
+  params: Params,
```

Comment:
I think it is the user's responsibility to decide whether `params` could be optional. It would be a logical error if the `params` is required but not provided. Mark `params` always optional force an extra check before use it, which is inconvenient and seems unnecessary.